cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java

```diff
@@ -594,7 +594,8 @@ public static MetadataResponse prepareMetadataResponse(List<Node> brokers,
     responseData.setClusterAuthorizedOperations(MetadataResponse.AUTHORIZED_OPERATIONS_OMITTED);
     topicMetadataList.forEach(topicMetadata -> responseData.topics().add(prepareMetadataResponseTopic(topicMetadata)));
 
-    return new MetadataResponse(responseData);
+    // In Kafka 2.8 we need to supply the Metadata response API version. In 2.5 this was version 9 and version 11 in 2.8.
+    return new MetadataResponse(responseData, (short) 9);
```

Comment:
Similar comment as above -- Would it be possible to use reflection to check whether the expected constructor exists (i.e. pre- and post-2.8) and switch to the existing constructor here?